### PR TITLE
Compose services

### PR DESCRIPTION
**Pull request by ibuildthecloud**

### CODE REVIEW

**Comment by imikushin** [comment]
Interesting!

**Comment by imikushin** [comment]
I've put the following to `/var/lib/rancher/conf/cloud-config.d` dir and rebooted:
`nginx.yml`:
```yml
#compose
nginx:
  image: nginx
  links:
  - network
  restart: always
  net: host
```

Didn't work (nginx container didn't start). Here's what ended up in `cloud-config-processed.yml`:
```yml
#cloud-config
nginx:
  image: nginx
  links:
  - network
  net: host
  restart: always
ssh_authorized_keys: null
```
I guess `nginx` key should've been put under `services`.

**Comment by ibuildthecloud** [comment]
@imikushin I'll look into that.  I haven't fully tested all the permutations of configurable.  I'm going to merge this PR even though I know there are some bugs.  This way we can cut a rc3 and do more extensive testing.

**Comment by ibuildthecloud** [comment]
@imikushin check out https://github.com/rancherio/os-services.  You can change the URL RancherOS looks at so you can fork that repo and try to add new stuff.  If you do `rancherctl service list` it should list what is in index.yml

We're working on docs (and a real docs website)

**Comment by deniseschannon** [comment]
@ibuildthecloud  the `rancherctl service list` doesn't work in v0.3.0-rc3.

I get this error. 
```
ERRO[0000] Failed to load https://raw.githubusercontent.com/rancherio/os-services/master/index.yml: non-200 http response: 404
disabled ubuntu-console
```

Logged a bug: https://github.com/rancherio/os/issues/187

File: docker/services.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancherio/os/config"
+	"github.com/rancherio/os/util"
+	"github.com/rancherio/rancher-compose/project"
+)
+
+type configEnvironemnt struct {
```
Comment (djmaze) [inline docker/services.go]: Typo

Comment (ibuildthecloud) [inline docker/services.go]: thanks, that's the worse when you have typos in type names.

**Pull request merged**